src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -127,9 +127,6 @@ private async Task UpdateConnectionAsync()
                 await Core.WebServiceHelper.SafeServiceCallAsync(() => sonarQubeService.ConnectAsync(connectionInformation,
                     CancellationToken.None), this.logger);
             }
-
-            Debug.Assert((boundProject != null) == sonarQubeService.IsConnected,
-                $"Inconsistent connection state: Solution bound={boundProject != null}, service connected={sonarQubeService.IsConnected}");
```

Comment:
This assertion was incorrect; we do have binding information, but could not connect.

I've manually check the TaintPackage load bug doesn't repro, and didn't find any issues apart from this assertion.

The message logging for the other features (suppressions, taint issues etc) looks pretty good e.g. along the lines of "couldn't update suppressions - server is not connected".